### PR TITLE
Add a check that warmup_setps is either 0 or >= 1

### PR DESCRIPTION
**Pull request by ymoslem**
# What does this PR do?

Update training_args.py to add a check that warmup_setps is either 0 or >= 1. Otherwise, raise an error.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

- trainer: @muellerzr and @pacman100

### CODE REVIEW

**Comment by ymoslem** [comment]
Hello! I am not sure what I should do to fix these errors in the tests. Thanks!

```
FAILED tests/models/dpt/test_modeling_dpt_auto_backbone.py::DPTModelTest::test_batching_equivalence - AssertionError: tensor(False) is not true : Batched and Single row outputs are not equal in DPTForDepthEstimation for key=predicted_depth. Difference=1.0.
================= 1 failed, 577 passed, 935 skipped in 56.07s ==================
```


```
#!/bin/bash -eo pipefail
python3 .circleci/parse_test_outputs.py --file tests_output.txt --fail

1 failed because `AssertionError: tensor(False) is not true ` -> Batched and Single row outputs are not equal in DPTForDepthEstimation for key=predicted_depth. Difference=1.0.
Number of failed tests: 1

Exited with code exit status 1

CircleCI received exit code 1
```

**Review by muellerzr — APPROVED** [review]
Thanks for this check! I think the test failures are unrelated to this. cc @amyeroberts for final review :)

**Comment by muellerzr** [comment]
You can try rebasing from `main`

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -1797,6 +1797,9 @@ def __post_init__(self):
                 " during training"
             )
 
+        if not isinstance(self.warmup_steps, int) or self.warmup_steps < 0:
```
Comment (amyeroberts) [inline src/transformers/training_args.py:1800]:
```suggestion
        if not isinstance(self.warmup_steps, int) or self.warmup_steps < 0 or 0 < self.warmup_steps <= 1:
```

**Review by amyeroberts — APPROVED** [review]
Thanks for adding this check! 

I think we can add another bound, to make sure the error message aligns with what's being tested

**Pull request merged**